### PR TITLE
[DS3] quantize w-uk,w-uv to fp8

### PR DESCRIPTION
**Pull request by gxd3**
[DeepSeekV3] quantize w-uk,w-uv weights to fp8.

This saved ~1 GB HBM per shard.
before this change: per device HBM usage after weight loading: 54.39 GB
with this change: per device HBM usage after weight loading: 53.46 GB

No performance impact on per-step time (matmul relevant to w-uk,w-uv is very small percentage of one layer's execution time). 

# Tests
Quality verified:
```
python ./tpu-inference/scripts/vllm/benchmarking/benchmark_serving.py --backend vllm --model deepseek-ai/DeepSeek-R1 --dataset-name mmlu --dataset-path /home/gxd_google_com/work-dir/mmlu/data/test --num-prompts 5000 --run_eval --temperature 0

============ Serving Benchmark Result ============
Successful requests:                     5000      
Benchmark duration (s):                  164.14    
Total input tokens:                      1015147   
Total generated tokens:                  10000     
Request throughput (req/s):              30.46     
Output token throughput (tok/s):         60.92     
Total token throughput (tok/s):          6245.66   
---------------Time to First Token----------------
Mean TTFT (ms):                          83540.16  
Median TTFT (ms):                        76951.81  
P99 TTFT (ms):                           162474.97 
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          1206.80   
Median TPOT (ms):                        1217.27   
P99 TPOT (ms):                           1238.46   
---------------Inter-token Latency----------------
Mean ITL (ms):                           1206.80   
Median ITL (ms):                         1217.27   
P99 ITL (ms):                            1238.46   
----------------End-to-end Latency----------------
Mean E2EL (ms):                          84746.95  
Median E2EL (ms):                        78172.75  
P99 E2EL (ms):                           163539.41 
==================================================
Evaluating MMLU...
[nltk_data] Downloading package punkt to
[nltk_data]     /home/gxd_google_com/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package punkt_tab to
[nltk_data]     /home/gxd_google_com/nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!

Results
{'accuracy': 0.8752, 'gen_num': 5000}
```

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/layers/vllm/backends/flash_attn_mla.py [hunk 1]
```diff
@@ -117,13 +117,18 @@ def forward(self, q: torch.Tensor, kv_c_normed: torch.Tensor,
...
+        input_dtype = q.dtype
 
         # Prepare inputs
         q_nope, q_pe = jnp.split(q, [self.qk_nope_head_dim], axis=2)
 
         # (B, N, P) x (N, P, L) -> (B, N, L)
         # torch nn param
-        q_nope = jnp.einsum("bnp,npl->bnl", q_nope, jax_view(layer.W_UK_T))
+        q_nope = (jnp.einsum("bnp,npl->bnl",
+                             q_nope,
```
Comment (kyuyeunk) [inline tpu_inference/layers/vllm/backends/flash_attn_mla.py:128]: should we also quantize activation (`q_nope`) so it will speed up compute? or does it cause some numeric issues?

Comment (gxd3) [reply tpu_inference/layers/vllm/backends/flash_attn_mla.py:128]: I have tried that approach, i.e.
```
q_nope, q_nope_scale = quantize_tensor(to_jax_dtype(layer.W_UK_T.dtype),q_nope)
q_nope_scale = q_nope_scale[..., None]
q_nope = jnp.einsum("bnp,npl->bnl", q_nope, jax_view(layer.W_UK_T), preferred_element_type=jnp.float32)
q_nope *= q_nope_scale
q_nope *= jax_view(layer.W_UK_T_scale)
q_nope = q_nope.astype(jnp.bfloat16)
```
But there isn't obserable speedup from it (actually slower due to the overhead of quantizing q_nope)

Comment (kyuyeunk) [reply tpu_inference/layers/vllm/backends/flash_attn_mla.py:128]: okay got it. also, using `quantize_tensor` during runtime is probably not optimal because it has bunch of "nice to have features" - which can be removed without hurting accuracy if you know what you are doing.
